src/components/versions/versions.provider.js

```diff
@@ -14,44 +10,63 @@ class VersionsService {
   }
 
   getInformations (clientUrl = this.$location.absUrl()) {
-    var clientUrlSplit = clientUrl.split('#')
-    var leftPartSplit = clientUrlSplit[0].split('/')
-
-    if (isEmpty(last(leftPartSplit))) {
-      leftPartSplit = initial(leftPartSplit)
+    const PROD_REGEX = /https?:\/\/(dev_)?([\w\-_]+)\.ui-kit\.ovh/
+
+    let branch = 'local'
+    let environment = 'localhost'
+    let version = 'latest'
+    let params = ''
+    const matches = PROD_REGEX.exec(clientUrl)
+    const splitUrl = clientUrl.split('#')
+
+    if (matches && matches.length === 3) {
+      environment = matches[1] === 'dev_' ? 'dev' : 'prod'
+      branch = matches[2]
+      version = branch === 'master' ? 'latest' : branch
     }
 
-    var version = 'latest'
-
-    if (leftPartSplit.length > 3) {
-      version = last(leftPartSplit)
-      leftPartSplit = initial(leftPartSplit)
+    if (splitUrl.length === 2) {
+      params = `#${splitUrl[1]}`
     }
 
     return {
-      rightPart: clientUrlSplit.length === 2 ? `#${clientUrlSplit[1]}` : null,
-      leftPartWithoutVersion: leftPartSplit.join('/'),
-      leftPartWithVersion: clientUrlSplit[0],
-      version: version
+      branch,
+      environment,
+      version,
+      params
     }
   }
 
-  getNewVersionUrl (newVersion, clientUrl = this.$location.absUrl()) {
-    var informations = this.getInformations(clientUrl)
+  getUrl (environment, branch, path) {
+    if (!path) {
+      path = ''
+    }
```

Comment:
You can use the ES6 default parameter
`getUrl (environment, branch, path = '') { ... }`